### ISSUE TITLE
Tags are shown as releases

### ISSUE THREAD

**Issue by rEnr3n** [meiji163/gh-notify#32]
I ran `gh notify` today and I got this errors:

```
$ gh notify
.gh: Not Found (HTTP 404)
/home/ren/.local/share/gh/extensions/gh-notify/gh-notify: line 147: $'  \E[1;34m"message"\E[m\E[1;38m:\E[m \E[32m"Not Found"\E[m\E[1;38m,\E[m': command not found
gh: Not Found (HTTP 404)
/home/ren/.local/share/gh/extensions/gh-notify/gh-notify: line 147: $'  \E[1;34m"message"\E[m\E[1;38m:\E[m \E[32m"Not Found"\E[m\E[1;38m,\E[m': command not found
```

I let the command run for a few seconds and the interactive ui showed up. I saw in my notifications two entries from jetpack-io/devbox. The third column shows `Release {` which is unusual. I opened github on my browser and checked the notifications there but there are no entries for devbox there. I checked the devbox repo and there is no release named 0.1.2-dev but there is a tag; a single tag not two. I marked all (except devbox) notifications as read on my browser. I run `gh notify` again and I don't see the error anymore. The devbox tag didn't show up either.

![2022-11-29_09:47:53+0800](https://user-images.githubusercontent.com/18730618/204420265-725e3ec9-fd21-4e48-875e-b3f40e025bef.png)

**Comment by LangLangBart** [comment]
Something nice for troubleshooting, thanks for the report. Haven't come across this problem yet.

Could you paste the following into your terminal and pass on the response?
```zsh
gh api -X GET notifications  -f all="true" -f page=1 -f per_page=50 --jq '.[] | select(.repository.full_name | contains ("jetpack"))'
```
It should return all array elements of the last 50 notifications which contain `jetpack` in the `.repository.full_name` field.

---

the `404` response from GitHub indicates that the release was not found.
```json
"message": "Not Found"
```
I assumed that a version might have been created and then deleted, but a test on a private repository did not confirm my hunch.


Currently I don't know the reason for the problem, in the meantime an additional check could be added to test if the last `gh` command in the code below was successful. 

https://github.com/meiji163/gh-notify/blob/6347de2493c38643d1a1500a80fb85dc7540526b/gh-notify#L127-L131

**Comment by LangLangBart** [comment]
@rEnr3n I was able to reproduce it, by passing `null` as the `"$url"` in line 129.

```diff
--- a/gh-notify
+++ b/gh-notify
@@ -126,7 +126,10 @@ print_notifs() {
                     number=$(basename "$url" | head -c 7)
                 elif grep -q "Release" <<<"$type"; then
                     release_info=()
+                    set -x
+                    url=null
                     while IFS='' read -r line; do release_info+=("$line"); done < <(gh api --cache=20s "$url" --jq '.tag_name, .prerelease')
+                    set +x
                     number="${release_info[0]}"
                     "${release_info[1]}" && type="Pre-release"
                 else
```

```zsh
❯ gh notify -s -n1
.++ url=null
++ IFS=
++ read -r line
+++ gh api --cache=20s null --jq '.tag_name, .prerelease'
gh: Not Found (HTTP 404)
++ release_info+=("$line")
++ IFS=
++ read -r line
++ release_info+=("$line")
++ IFS=
++ read -r line
++ release_info+=("$line")
++ IFS=
++ read -r line
++ release_info+=("$line")
++ IFS=
++ read -r line
++ set +x
/Users/paria/.local/share/gh/extensions/gh-notify/gh-notify: line 153:   "message": "Not Found",: command not found

29/Nov 06:31  LangLangBart/test  Release { ●  cool
```

<img src="https://user-images.githubusercontent.com/92653266/204463532-30deaa3b-3aa1-41a5-b651-10b6d7970278.png" width="600">

---

For some reason the `"$url"` of a `Release` type can be `null`, adding a null check should be sufficient.

**Comment by LangLangBart** [comment]
The problem occurs when a user creates a release and then deletes the tag number. 
```zsh
# create a tag
git tag v2
# Push it
git push origin v2

# create the release
gh release create ...

# delete the tag
git push origin --delete tag v2
```

---

@rEnr3n the information I asked for is no longer necessary. The cause of the problem is clear and should be resolved with the associated PR.

**Comment by rEnr3n** [comment]
I'm not too familiar with github's releases and tags. But in the case of the devbox repo, is that what exactly happened (create release and delete tag)? I don't see that release on their repo but there is one tag.

**Comment by LangLangBart** [comment]
if you are in contact with them, it would be good to get clarification.

When I delete the tag, the release is converted to a draft in my tests.

**Comment by LangLangBart** [comment]
### Reopening
the problem was not solved, a URL link was pulled for a `release` event, but the content cannot be retrieved.

Below are three `release` notifications for `jetpack-io/devbox`, but only the third one works properly.

```zsh
gh api -X GET notifications -f all="true" -f page=1 -f per_page=50 | jq '.[] | select(.repository.full_name | contains ("jetpack")) | .subject'

{
  "title": "0.1.3-dev",
  "url": "https://api.github.com/repos/jetpack-io/devbox/releases/85819422",
  "latest_comment_url": "https://api.github.com/repos/jetpack-io/devbox/releases/85819422",
  "type": "Release"
}
{
  "title": "0.1.3-dev",
  "url": "https://api.github.com/repos/jetpack-io/devbox/releases/85402520",
  "latest_comment_url": "https://api.github.com/repos/jetpack-io/devbox/releases/85402520",
  "type": "Release"
}
{
  "title": "0.1.2",
  "url": "https://api.github.com/repos/jetpack-io/devbox/releases/85248378",
  "latest_comment_url": "https://api.github.com/repos/jetpack-io/devbox/releases/85248378",
  "type": "Release"
}
```

**Comment by LangLangBart** [comment]
Fixed with #36
